libraries/tock-register-interface/src/macros.rs

```diff
@@ -51,13 +51,79 @@ macro_rules! register_bitmasks {
     {
         $valtype:ty, $reg_desc:ident, $(#[$outer:meta])* $field:ident,
                     $offset:expr, $numbits:expr,
-                    [$( $(#[$inner:meta])* $valname:ident = $value:expr ),*]
+                    [$( $(#[$inner:meta])* $valname:ident = $value:expr ),+]
     } => {
         #[allow(non_upper_case_globals)]
         #[allow(unused)]
         pub const $field: Field<$valtype, $reg_desc> =
             Field::<$valtype, $reg_desc>::new((1<<($numbits-1))+((1<<($numbits-1))-1), $offset);
 
+        #[allow(non_snake_case)]
+        #[allow(unused)]
+        $(#[$outer])*
+        pub mod $field {
+            #[allow(unused_imports)]
+            use $crate::registers::{FieldValue, TryFromValue};
+            use super::$reg_desc;
+
+            $(
+            #[allow(non_upper_case_globals)]
+            #[allow(unused)]
+            $(#[$inner])*
+            pub const $valname: FieldValue<$valtype, $reg_desc> =
+                FieldValue::<$valtype, $reg_desc>::new((1<<($numbits-1))+((1<<($numbits-1))-1),
```

Comment:
This seems like an obscure arithmetic expression which would IMO deserve its own macro to clarify what it does, given that it's used in many places now.

Reply:
Also, alternative proposal that may be more readable: `!0 >> (core::mem::size_of::<$valtype>() * 8 - $numbits)`